### PR TITLE
RNMT-3424 Add support for Dark Mode

### PR DESCRIPTION
**Pull request by EiyuuZack**
## Description

This PR adds support for dark mode.

## Context

References https://outsystemsrd.atlassian.net/browse/RNMT-3424

## Type of changes

- [x] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [ ] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Core
- [ ] Public API
- [x] Demo App

## Tests

Manual testing -- see below

## Screenshots (if appropriate)

**iOS 12**
cardio_ios12

**iOS 13**
cardio_ios13

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

**Review by usernuno — COMMENTED** [review]
Most files don't import the CardIOMacros.h, isn't it required in all of them?

**Comment by EiyuuZack** [comment]
> Most files don't import the CardIOMacros.h, isn't it required in all of them?

Some .m files already imported it, or they respective .h did. Only had to import it in the one file you see here.

File: Classes/CardIODataEntryViewController.m [hunk 1]
```diff
@@ -164,7 +164,14 @@ - (void)viewDidLoad {
     self.cardView.bounds = CGRectZeroWithSize(CGSizeMake((CGFloat)ceil(self.floatingCardView.bounds.size.width),
                                                          (CGFloat)ceil(self.floatingCardView.bounds.size.height)));
     self.cardView.contentMode = UIViewContentModeScaleAspectFit;
-    self.cardView.backgroundColor = kColorViewBackground;
+    
+    if (iOS_13_PLUS) {
```
Comment (luissilvaos) [inline Classes/CardIODataEntryViewController.m:168]: Is this a macro? You could have used the [Availability API](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/marking_api_availability_in_objective-c) for this use case.

**Review by luissilvaos — DISMISSED** [review]

**Review by pmaroco — DISMISSED** [review]

File: Classes/CardIODataEntryViewController.m (hunk 1 above)
Comment (EiyuuZack) [reply Classes/CardIODataEntryViewController.m:168]: Considered it, but upstream already had this nifty macro setup that they use everywhere. When in Rome...

**Review by usernuno — DISMISSED** [review]

File: Classes/CardIODataEntryViewController.m (hunk 1 above)
Comment (luissilvaos) [reply Classes/CardIODataEntryViewController.m:168]: Yeah, I only noticed that later and that was the reason why your PR got approved 😄

**Review by Chuckytuh — DISMISSED** [review]

**Review by Chuckytuh — DISMISSED** [review]

**Review by usernuno — DISMISSED** [review]

**Review by usernuno — APPROVED** [review]

**Review by Chuckytuh — APPROVED** [review]

**Pull request merged**